### PR TITLE
Replace Travis CI with a Github Actions workflow to run builds

### PR DESCRIPTION
**Pull request by kategengler**
This recreates our current CI from Travis on GitHub Actions. 

Currently Github Actions does not support any yaml reuse, so it is a bit verbose, but supposedly they will support in the future. 

Had to be a bit creative to re-create the cron builds for beta, release, etc. because Actions only currently supports scheduled builds on the default branch. 

For this to work, we need to set up a number of secrets in the repo:

- NPM_AUTH_TOKEN, Token for publishing to npm
- S3_SECRET_ACCESS_KEY, For the bucket where we deploy channels
- S3_ACCESS_KEY_ID
- PERSONAL_TOKEN, Needs to be a Github token with `repo` access to this repo, due to a limitation in actions that does not let a push from one workflow with the default token to trigger another workflow run

### CODE REVIEW

**Comment by kategengler** [comment]
It occurs to me that Travis should fail for this PR... 

You can see the build running and working here https://github.com/kategengler/ember.js/runs/769332062?check_suite_focus=true Except for the deploy to AWS which doesn't work due to permissions (and will work given correct tokens)

**Comment by rwjblue** [comment]
Doesn't look like any CI runs were kicked off here...

**Comment by kategengler** [comment]
The actions won't run until they are master, afaict.

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+
+jobs:
+  lint:
+    name: Linting
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v1
+        with:
+          node-version: 10.x
```
[the remark this answers is not in the document]
Comment (kategengler) [reply .github/workflows/build.yml:26]: I haven't personally used volta, so I'm not sure what is needed

**Comment by rwjblue** [comment]
If you push to a branch on the repo it should pick them up, but from your fork it won’t. 

But let’s remove the deletion of .travis.yml so we can have both while we shake out the actions.

**Comment by kategengler** [comment]
I'd need to revisit the deploy code to have travis deploys work the same time as this. Should I push to a branch on emberjs/ember.js so we can see the actions run?

**Comment by chancancode** [comment]
> Should I push to a branch on emberjs/ember.js so we can see the actions run?

Yes, assuming it won't do anything "dangerous" like publishing (which it shouldn't, since it haven't landed on master yet). We may want to selectively enable some of the things that usually only run on master on your branch (e.g. by adding it to the branches list) so we can test some of these things.

**Comment by kategengler** [comment]
It won't, but the CI can also be seen running on my fork kategengler/ember.js/runs/769332062?check_suite_focus=true

**Pull request closed without merging**